test/DatabaseTest.cc

```diff
@@ -208,1046 +251,174 @@ TEST_CASE_METHOD(DatabaseTest, "Database w/o config") {
-    CHECK(error.domain == kCBLDomain);
-    CHECK(error.code == kCBLErrorNotFound);
-}
-
+    bool useSHA256Key = true;
+    SECTION("SHA-265 Key")
+    {
+        CBLEncryptionKey_FromPassword(&key, "sekrit"_sl);
+    }
+    
+    SECTION("SHA-1 Key")
+    {
+        useSHA256Key = false;
+        CBLEncryptionKey_FromPasswordOld(&key, "sekrit"_sl);
+    }
+    
+    CBLDatabaseConfiguration config = {nullslice, key};
+    CBLDatabase *defaultdb = CBLDatabase_Open("encdb"_sl, &config, &error);
+    REQUIRE(defaultdb);
+    alloc_slice path = CBLDatabase_Path(defaultdb);
+    cerr << "Default database is at " << path << "\n";
+    CHECK(CBL_DatabaseExists("encdb"_sl, nullslice));
 
-TEST_CASE_METHOD(DatabaseTest, "Legacy - Purge Document") {
-    createDocument(db, "doc1", "foo", "bar");
-    createDocument(db, "doc2", "foo", "bar");
+    CBLDatabaseConfiguration config1 = CBLDatabase_Config(defaultdb);
+    REQUIRE(config1.encryptionKey.algorithm  == key.algorithm);
+    REQUIRE(memcmp(config1.encryptionKey.bytes, key.bytes, 32) == 0);
     
-    CBLError error;
-    const CBLDocument* doc = CBLDatabase_GetDocument(db, "doc1"_sl, &error);
-    REQUIRE(doc);
-    CHECK(CBLDocument_Sequence(doc) == 1);
+    // Correct key from config:
+    CBLDatabase *correctkeydb = CBLDatabase_Open("encdb"_sl, &config1, &error);
+    REQUIRE(correctkeydb);
+    CBLDatabase_Release(correctkeydb);
     
-    CHECK(CBLDatabase_PurgeDocument(db, doc, &error));
-    CBLDocument_Release(doc);
-    CHECK(!CBLDatabase_GetDocument(db, "doc1"_sl, &error));
+    // No key:
+    {
+        ExpectingExceptions x;
+        CBLDatabase *nokeydb = CBLDatabase_Open("encdb"_sl, nullptr, &error);
+        REQUIRE(nokeydb == nullptr);
+        CHECK(error.domain == kCBLDomain);
+        CHECK(error.code == kCBLErrorNotADatabaseFile);
+    }
     
-    CHECK(CBLDatabase_PurgeDocumentByID(db, "doc2"_sl, &error));
-    CHECK(!CBLDatabase_GetDocument(db, "doc2"_sl, &error));
+    // Wrong key:
+    {
+        ExpectingExceptions x;
+        CBLEncryptionKey key2;
+        
+        if (useSHA256Key) {
+            CBLEncryptionKey_FromPassword(&key2, "wrongpassword"_sl);
+        } else {
+            CBLEncryptionKey_FromPasswordOld(&key2, "wrongpassword"_sl);
+        }
+        
+        CBLDatabaseConfiguration config2 = {nullslice, key2};
+        CBLDatabase *wrongkeydb = CBLDatabase_Open("encdb"_sl, &config2, &error);
+        REQUIRE(wrongkeydb == nullptr);
+        CHECK(error.domain == kCBLDomain);
+        CHECK(error.code == kCBLErrorNotADatabaseFile);
+    }
+
+    CHECK(CBLDatabase_Delete(defaultdb, &error));
+    CBLDatabase_Release(defaultdb);
+    CHECK(!CBL_DatabaseExists("encdb"_sl, nullslice));
 }
 
+#endif
+
+#pragma mark - Save Document:
 
-TEST_CASE_METHOD(DatabaseTest, "Legacy - Purge Already Purged Document") {
-    createDocument(db, "doc1", "foo", "bar");
+TEST_CASE_METHOD(DatabaseTest, "Save Document into Different DB Instance") {
```

Comment:
Operation is performed on 2 different dbs default collection, rather than 2 collections within the same db. It is pretty much the same and if is still considered redundant then will remove.